content/en/blog/_posts/2023-01-17-confidential-kubernetes.md

```diff
@@ -0,0 +1,324 @@
+
+Although there is a performance overhead, it should be low enough to enable real-world workloads
+to run in these protected environments and improve the security and privacy of our data.
+
+## Confidential Computing compared to FHE, ZKP, and MPC
+
+Fully Homomorphic Encryption (FHE), Zero Knowledge Proof/Protocol (ZKP), and Multi-Party
+Computations (MPC) are all a form of encryption or cryptographic protocols that offer
+similar security guarantees to Confidential Computing but do not require hardware support.
+
+Fully (also partially and somewhat) homomorphic encryption allows one to perform
+computations, such as addition or multiplication, on encrypted data. This provides
+the property of encryption in use but does not provide integrity protection or attestation.
+
+Zero Knowledge Proofs or Protocols are a privacy-preserving technique (PPT) that
+allows one party to prove facts about their data without revealing anything else about
+the data. ZKP can be used instead of or in addition to Confidential Computing to protect
+the privacy of the involved parties and their data. Similarly, Multi-Party Computation
+enables multiple parties to work together on a computation, i.e, each party provides
+their data to the result, without leaking their data to any of the other parties.
+
+## Use cases of Confidential Computing
+
+The presented Confidential Computing platforms show that both the isolation of a single container
+process and therefore minimization of the trusted computing base, and the isolation of a
+full virtual machine are possible. This already enabled a lot of interesting and secure
+projects to emerge:
+
+### Confidential Containers
+
+The [Confidential Containers](https://github.com/confidential-containers) (CoCo) project enables users
+to process and run each container of a pod inside a confidential context.
+[Confidential Containers](https://www.cncf.io/projects/confidential-containers/) is a CNCF sandbox project.
+
+CoCo offers an extension to Kubernetes clusters that are comprised of
+nodes that support TEEs, to launch containerized workloads within a TEE created
+on the fly. With CoCo, the entire lifecycle of a pod, including the image handling occurs
+within a TEE leaving the cluster node outside of the TCB: for each pod, a TEE is launched, followed by the verification of the container images' signatures before pull and seeking any
+decryption keys to launch the container. The image policy and layer decryption keys a obtained from a trusted *key broker service* that involves a TEE remote attestation as part of the process. Coco's near term goals are to move to CSPs' Kubernetes
+clusters and further separate management and tenant APIs to restrict the
+Kubernetes cluster admin access to the tenant pods and the containers within.
+CoCo currently supports both AMD and Intel TEE solutions. It is defining both
+a generic Key Broker and Attestation service to support an end-to-end confidential
+flow.
```

Comment:
```suggestion
CoCo can be installed on a Kubernetes cluster with an operator.
The operator will create a set of runtime classes that can be used to deploy
pods inside an enclave on several different platforms, including
AMD SEV, Intel TDX, Secure Execution for IBM Z, and Intel SGX.

CoCo is typically used with signed and/or encrypted container images
which are pulled, verified, and decrypted inside the enclave.
Secrets, such as image decryption keys, are conditionally provisioned
to the enclave by a trusted Key Broker Service that validates the
hardware evidence of the TEE prior to releasing any sensitive information.

CoCo has several deployment models. Since the Kubernetes control plane
is outside the TCB, CoCo is suitable for managed environments. CoCo can
be run in virtual environments that don't support nesting with the help of an
API adaptor that starts pod VMs in the cloud. CoCo can also be run on
bare metal, providing strong isolation even in multi-tenant environments.
```